pkg/expect/expect.go

```diff
@@ -96,10 +97,16 @@ func (ep *ExpectProcess) read() {
 }
 
 // ExpectFunc returns the first line satisfying the function f.
-func (ep *ExpectProcess) ExpectFunc(f func(string) bool) (string, error) {
+func (ep *ExpectProcess) ExpectFunc(ctx context.Context, f func(string) bool) (string, error) {
 	i := 0
 
 	for {
+		select {
+		case <-ctx.Done():
+			return "", fmt.Errorf("failed to find match string: %w", ctx.Err())
+		default:
```

Comment:
We could consider merging this select with time.Sleep

Reply:
Tried to merge with time.Sleep and simplify a little bit.
Looks legitimately and units are passed (: